GitUp/Application/QuickViewModel.m

```diff
@@ -0,0 +1,151 @@
+//
+//  QuickViewModel.m
+//  Application
+//
+//  Created by Dmitry Lobanov on 15/09/2019.
+//
+
+@import GitUpKit;
+#import <GitUpKit/XLFacilityMacros.h>
+#import "QuickViewModel.h"
+
+@interface QuickViewModel ()
+
+@property (weak, nonatomic) GCLiveRepository *repository;
+
+@property (assign, nonatomic) NSUInteger index;
+@property (strong, nonatomic) NSMutableArray *commits;
+
+@property (strong, nonatomic) GCHistoryWalker *ancestors;
+@property (strong, nonatomic) GCHistoryWalker *descendants;
+
+#pragma mark - Protected
+- (void)loadMoreAncestors;
+- (void)loadMoreDescendants;
+@end
+
+@implementation QuickViewModel
+
+#pragma mark - Initialization
+- (instancetype)initWithRepository:(GCLiveRepository *)repository {
+  if ((self = [super init])) {
+    self.repository = repository;
+  }
+  return self;
+}
+
+#pragma mark - Loading
+- (void)loadMoreAncestors {
+  if (![_ancestors iterateWithCommitBlock:^(GCHistoryCommit* commit, BOOL* stop) {
+    [_commits addObject:commit];
+  }]) {
+    _ancestors = nil;
+  }
+}
+
+- (void)loadMoreDescendants {
+  if (![_descendants iterateWithCommitBlock:^(GCHistoryCommit* commit, BOOL* stop) {
+    [_commits insertObject:commit atIndex:0];
+    _index += 1;  // We insert commits before the index too!
+  }]) {
+    _descendants = nil;
+  }
+}
+
+#pragma mark - Checking
+- (BOOL)hasPrevious {
+  return _index + 1 < _commits.count;
+}
+
+- (BOOL)hasNext {
+  return _index > 0;
+}
+
+- (BOOL)hasPaging {
+  return _commits != nil;
+}
+
+// TODO: Rename them appropriately.
+// Blowing mind.
```

Comment:
Is this an actual TODO?

Reply:
Well, I added a note about it.

If we moving backward, then, we should increase index. (And load Ancestors)
If we moving forward, then, we should decrease index. (And load Descendants)

If these statements are logically correct, then, we could keep current API.
Otherwise, we need to rethink it.